mullvad-obfuscated-dns-proxy/src/forwarder/mod.rs

```diff
@@ -0,0 +1,108 @@
+use std::{io, task::{ready, Poll}};
+
+use tokio::{
+    io::{AsyncRead, AsyncWrite},
+    net::TcpStream,
+};
+
+use crate::config::Obfuscator;
+
+pub struct Forwarder {
+    read_obfuscator: Box<dyn Obfuscator>,
+    write_obfuscator: Box<dyn Obfuscator>,
+    server_connection: TcpStream,
+}
+
```

Comment:
Why does the `Forwarder` use dynamic dispatch for the `Obfuscator`? Would it not be as simple to use generics instead?